### ISSUE TITLE
[Feature Request]: Compatbility with scripts that move files when a replay buffer or recording is saved

### ISSUE THREAD

**Issue by Pandaptable** [DmitriySalnikov/OBSNotifier#21]
### Feature description

Self explanatory, example script: https://github.com/padiix/OBS-Recordings-Organizer

### Implementation Ideas

I have a basic implementation but my knowledge of C# is not really good enough to finish it by adding an option for it in the UI and such.. that can be found here: https://github.com/Pandaptable/OBSNotifier

I'm sure there's better ways to implement this but this works for me personally, however I think a proper implementation may be useful for others.

**Comment by DmitriySalnikov** [comment]
> I have a basic implementation but my knowledge of C# is not really good enough to finish it by adding an option for it in the UI

In fact, the settings interface in this program is a mess (UI and code). And I'd like to rewrite everything.

> I'm sure there's better ways to implement this but this works for me personally, however I think a proper implementation may be useful for others.

There is no exit from `while` in your code if the file name changes or it is not created. Also I don't like `async` inside `async` (although it may be a normal practice in C# 🤷‍♂️).

But I really like the idea of replacing the subfolder with the name of the application. I would like to see if this can be done more reliably and if it is even possible to replace the path to the file before saving it to OBS.

**Comment by Pandaptable** [comment]
yeah no my code is very very bad lmao, i just threw it together and it works flawlessly for me

it was more so to get one working for me until a potential implementation by you

**Comment by Pandaptable** [comment]
note: I personally use a modified version of the example script I provided where I can overwrite exe names with any subfolder i want (inside of a Replays folder)
